ortools/linear_solver/xpress_interface.cc

```diff
@@ -1247,36 +1411,30 @@ MPSolver::ResultStatus XpressInterface::Solve(MPSolverParameters const& param) {
 
   // Extract the model to be solved.
   // If we don't support incremental extraction and the low-level modeling
-  // is out of sync then we have to re-extract everything. Note that this
-  // will lose MIP starts or advanced basis information from a previous
-  // solve.
+  // is out of sync then we have to re-extract everything.
   if (!supportIncrementalExtraction && sync_status_ == MUST_RELOAD) Reset();
   ExtractModel();
   VLOG(1) << absl::StrFormat("Model build in %.3f seconds.", timer.Get());
 
-  // Set log level.
-  XPRSsetintcontrol(mLp, XPRS_OUTPUTLOG, quiet() ? 0 : 1);
+  // Enable log output.
+  if (!quiet())
+    XPRSaddcbmessage(mLp, cbmessage, &std::cout, 0);
```

Comment:
You may want to pass the logger here instead of `std::cout`. The reason for that is that `std::cout` isn't always captured by applications that use OR-Tools.

It would be best to let the user decide where they want to place the solver output, but I'm not aware of that possibility.

Reply:
Instead of passing the logger, I now pass `nullptr` and in function `cbmessage` write to `LOG(INFO)`, `LOG(WARNING)`, LOG(ERROR)`, depending on the message type (commit 8dfa1e5f0302d1a5a307ae4f27a11439f339141f).

Reply:
Excellent ! GLog offer a [`LogSink` virtual class](https://github.com/google/glog/blob/674283420118bb919f83ceb3d9dee31ef43ff3aa/src/glog/logging.h.in#L1582), applications can override it and use `AddLogSink` to handle logs from GLog.